### PR TITLE
Script to clean external cluster

### PR DESCRIPTION
**Pull request by alejandroEsc**
Script added to had directory, including additional comments on Readme file. This should go away once clean up is merged, but for now this should save us all time.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -139,3 +139,35 @@ index 8fac530..3d6c246 100644
 <a name="kvm2">1</a> If using minikube on linux, you may prefer to use the
 [kvm2 driver](https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#kvm2-driver).
 To do so, add the `--vm-driver=kvm2` flag after installing the driver.
+
+## Cleaning External Cluster for Dev Work
+There is a tool to help you dev quickly and redeploy with clean states, the script is located in the hack directory.
```
Comment (guineveresaenger) [inline README.md]: s/dev/develop 
can you link to the script location?

File: README.md [hunk 2]
```diff
@@ -139,3 +139,35 @@ index 8fac530..3d6c246 100644
...
+First a few assumptions:
+
+- Using pivoting.
```
Comment (guineveresaenger) [inline README.md]: Is "pivoting" explained anywhere? I don't really know what this means.

File: README.md [hunk 3]
```diff
@@ -139,3 +139,35 @@ index 8fac530..3d6c246 100644
...
 [kvm2 driver](https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#kvm2-driver).
 To do so, add the `--vm-driver=kvm2` flag after installing the driver.
+
+## Cleaning External Cluster for Dev Work
+There is a tool to help you dev quickly and redeploy with clean states, the script is located in the hack directory.
+First a few assumptions:
+
+- Using pivoting. 
+- Using an external cluster such as minikube, or your own.
+- minikube/external cluster has kubeconfig file either exported in env variable or in the default directory.
```
Comment (guineveresaenger) [inline README.md]: are there instructions for handling this kubeconfig file?

File: README.md (hunk 2 above)
Comment (alejandroEsc) [reply README.md]: its a clusterctl concept.

File: README.md (hunk 1 above)
Comment (alejandroEsc) [reply README.md]: location is on the readme entry

File: README.md (hunk 3 above)
Comment (alejandroEsc) [reply README.md]: Nothing special here, just how you would normally use a kubeconfig file for your cluster

File: README.md (hunk 2 above)
Comment (davidewatson) [reply README.md]: Why not remove these all of these assumptions? Notes can be more valuable to developers when they are terse. If there are are requirements, then try to represent them as code (see below).

We need another document (maybe the Provider Implementors Guide?), upstream, which describes concepts like external, internal, pivoting, etc.

File: README.md [hunk 4]
```diff
@@ -139,3 +139,35 @@ index 8fac530..3d6c246 100644
...
+So first create a cluster as part of your dev cycle
+
+```
+$ ./bin/clusterctl create cluster -v 7 --provider ssh -c ./clusterctl/examples/ssh/out/cluster.yaml -m ./clusterctl/examples/ssh/out/machines.yaml -p ./clusterctl/examples/ssh/out/provider-components.yaml --existing-bootstrap-cluster-kubeconfig=minikube.kubeconfig
+```
+Notice that we are using the `--existing-bootstrap-cluster-kubeconfig` flag, that way we do not have to recreate minikube over and over which takes a very long time.
+Now, we have finished, pivoted, and completed (whether succeeded or not), next step is to just clean our cluster for the next cycle:
+
+```
+$./hack/delete_deployments.sh ssh-controlplane-c97bl new-test-1
```
Comment (davidewatson) [inline README.md]: If the assumptions above are removed then we should export our `KUBECONFIG` here in order to be explicit:
```
export KUBECONFIG=minikube.kubeconfig
```

File: README.md (hunk 1 above)
Comment (davidewatson) [reply README.md]: These are reasonable requests.

File: README.md (hunk 2 above)
Comment (guineveresaenger) [reply README.md]: rephrase: I've some idea about pivoting, but a newcomer to this repository/clusterctl would not. Leaving links and breadcrumbs is helpful.

**Pull request merged**